static/js/message_scroll.js

```diff
@@ -70,9 +70,10 @@ exports.show_end_of_results_notice = function () {
     // It's a bit hacky to use the href, but
     // !filter.includes_full_stream_history() implies streams:public
     // wasn't already present.
-    $(".all-messages-search-caution a.search-shared-history").attr(
-        "href", window.location.hash.replace("#narrow/", "#narrow/streams/public/")
-    );
+    const operators = narrow_state.filter().operators();
+    const current_hash = hash_util.operators_to_hash(operators);
+    const new_hash = "#narrow/streams/public" + current_hash.substring(7);
```

Comment:
Let's not hardcode 7 here; and would it work to just add `streams: public` to the operators list we're passing into `operators_to_hash`?